packages/immutable-arraybuffer/index.js

```diff
@@ -160,7 +167,8 @@ export const transferBufferToImmutable = (buffer, newLength = undefined) => {
       }
     }
   }
-  return new ImmutableArrayBufferInternal(buffer);
+  const result = new ImmutableArrayBufferInternal(buffer);
+  return /** @type {ArrayBuffer} */ (/** @type {unknown} */ (result));
```

Comment:
I added this cast to both, so that the external typing is always simply `ArrayBuffer`. With these cast here, most messy types were no longer needed.